barretenberg/cpp/src/barretenberg/plonk_honk_shared/composer/permutation_lib.hpp

```diff
@@ -384,6 +384,13 @@ void compute_permutation_argument_polynomials(const typename Flavor::CircuitBuil
             compute_monomial_and_coset_fft_polynomials_from_lagrange<Flavor::NUM_WIRES>("id", key);
         }
     } else if constexpr (IsUltraFlavor<Flavor>) { // any UltraHonk flavor
+        // Allocate sigma and id polynomials
+        for (auto& sigma : key->polynomials.get_sigmas()) {
+            sigma = typename Flavor::Polynomial(key->circuit_size);
```

Comment:
wasn't sure if I could restrict any of these

Reply:
could restrict to the non-dyadic circuit size potentially? Unsure because computing them iterates over the EvaluationDomain...